app/src/main/kotlin/com/softteco/template/ui/feature/resetPassword/ResetPasswordViewModel.kt

```diff
@@ -0,0 +1,115 @@
+import com.softteco.template.ui.feature.ValidateFields.isHasMinimum
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class ResetPasswordViewModel @Inject constructor(
+    private val repository: ProfileRepository,
+) : ViewModel() {
+    private val loading = MutableStateFlow(false)
+    private val resetPasswordState = MutableStateFlow(false)
+    private var passwordStateValue = MutableStateFlow("")
+    private var snackBarState = MutableStateFlow(SnackBarState())
+
+    val state = combine(
+        loading,
+        passwordStateValue,
+        snackBarState,
+    ) { loading, passwordValue, snackBar ->
+        State(
+            loading = loading,
+            passwordValue = passwordValue,
+            fieldStatePassword = when {
+                passwordValue.isEmpty() -> PasswordFieldState.Empty
+                else -> PasswordFieldState.Success
+            },
+            isPasswordHasMinimum = passwordValue.isHasMinimum(),
+            isPasswordHasUpperCase = passwordValue.isHasCapitalizedLetter(),
+            snackBar = snackBar,
+            dismissSnackBar = { snackBarState.value = SnackBarState() },
+            onPasswordChanged = { passwordStateValue.value = it },
+            onResetPasswordClicked = ::onResetPassword
+        )
+    }.stateIn(
+        viewModelScope,
+        SharingStarted.Lazily,
+        State()
+    )
+
+    private fun handleError(error: ErrorEntity) {
+        val textId = when (error) {
+            ErrorEntity.AccessDenied -> R.string.error_example
+            ErrorEntity.Network -> R.string.error_example
+            ErrorEntity.NotFound -> R.string.error_example
+            ErrorEntity.ServiceUnavailable -> R.string.error_example
+            ErrorEntity.Unknown -> R.string.error_example
+        }
+        snackBarState.value = SnackBarState(
+            textId = textId,
+            show = true,
+        )
+    }
+
+    private fun onResetPassword() {
+        loading.value = true
+        val isAllFieldsValid = state.value.run {
+            fieldStatePassword is PasswordFieldState.Success &&
+                isPasswordHasUpperCase && isPasswordHasMinimum
+        }
+        if (isAllFieldsValid) {
+            viewModelScope.launch {
+                val resetPasswordDto = ResetPasswordDto(
+                    token = "token",
+                    password = passwordStateValue.value,
+                    confirmPassword = passwordStateValue.value,
+                )
+                when (val result = repository.resetPassword(resetPasswordDto)) {
+                    is Result.Success ->
+                        resetPasswordState.value =
+                            true // TODO: if success - go to profile screen
```

Comment:
It seems that the comment is unnecessary, since the implementation of the screen already implies a navigating to the Log in screen

Reply:
fixed